rpc/core/pipe.go

```diff
@@ -49,6 +49,14 @@ type peers interface {
 	Peers() p2p.IPeerSet
 }
 
+// Mempool extends the standard Mempool interface to allow getting
+// total txs size. ReapMaxTxs is used by UnconfirmedTxs to reap N transactions.
+type Mempool interface {
```

Comment:
Still reading through the code but are there are several implementations of this interface? 

OK, seems like there is only one implementation. Namely, `CListMempool` which sits in the `mempool` package. Shouldn't this interface be in the mempool package then?
Or shouldn't `ReapMaxTxs(n int) types.Txs` be part of the standard mempool interface? Looks like any actual (non-mock) mempool implementation has it anyways? At least it is repeated in the `MempoolWithWait` interface.

Reply:
It's confusing

```go
type MempoolWithWait interface {
	Mempool
	TxsFront() *clist.CElement
	TxsWaitChan() <-chan struct{}
	ReapMaxTxs(n int) types.Txs
}
```
and 
```go
type Mempool interface {
	mempl.Mempool
	ReapMaxTxs(n int) types.Txs
	TxsBytes() int64
}
```

As mentioned above `ReapMaxTxs` should live in the standard mempool. OR, if for some reason we want to keep the standard mempool interface small. Then, introduce yet another interface, e.g.

```go
type TxReaper interface {
  ReapMaxTxs(n int) types.Txs
}
```
and include this in both interfaces?